### DISCUSSION TITLE
Launcher is Causing TimeWait to IP 72.21.91.29

### DISCUSSION THREAD

**Discussion by JamiB1979** [nextgenhealthcare/connect#4828] [Q&A] [answered]
Hi all. Hope someone can help out. I've installed Mirth on three systems at a customer site. It works fine on another system for this customer. The two systems I have issue with are doing something very, very odd. I can start the Mirth Connect service, no issue. All is well. Within a min of the starting the launcher I get probably a hundred plus ports going into time wait for trying to connect to port 80 on address 72.21.91.29. I shut all of Mirth down and the time wait ports go away within a couple mins. This only happens when I start the launcher. 

Has anyone seen this before? Is this because the launcher is looking for an update? With all the ports in timewait, I can't establish new connections and actually get it to connect to other interfaces. 

Thanks in advance for any help!

**Comment by pacmano1** [comment]
See https://github.com/nextgenhealthcare/connect/issues/4807.

**Reply by JamiB1979** [reply]
Thanks so much! I gave this a go and it's still happening even after I blocked launcher.exe and launch.exe. Are there other files I'm maybe missing?

**Comment by pacmano1** [comment]
I think someone mentioned creating a local hosts file entry to just override DNS.
e.g.
```
127.0.0.1 ocsp.digicert.com
```

**Reply by JamiB1979** [reply]
I guess I'm not understanding what the DNS override is for? Sorry to sound stupid. I haven't had this issue with any other launchers so this is the first I'm hearing of this.

**Reply by pacmano1** [reply]
You should be seeing it on other launchers I think.  I am just sharing how to stop the traffic outbound through your firewalll by lying to the launcher on your workstation.   The server engine has nothing to do with this.   Although that engine does reach out to the internet as well.

**Reply by JamiB1979** [reply]
I thank you so much for your help. :-) I'm sorry I probably sounded rude there and that wasn't my intent and I'm truly sorry. That wouldn't affect anything else trying to go outbound would it? I'm just making sure since this is a customer system and not my own. 

Again I am really grateful. :-)

**Reply by pacmano1** [reply]
I would just let the launcher do what it does for now and wait to see if NG fixes the OSCP cert check described in the bug I referenced earlier.  i.e. "it is what it is".    If you are having problems with the launcher due to the customer's firewall blocking stuff, I would ask them to whitelist for now.    I don't worry about it on the many engines I manage because it is my workstation making those connections to the internet, not the server at the client site.   Meaning I am remote, so they don't see any of that.

**Reply by JamiB1979** [reply]
The customer is a tiny little site and they don't have any big corporate firewall so it's not blocking any traffic thankfully. The biggest issue is when it happens, the time wait ports won't let anything else on the server connect so I can't enable the channels to get data flowing. :-( Luckily I had one up and going before the bug appeared so I have that one. Also thankfully these interfaces don't need to be live for a couple months so we have time, I just have to get these channels up eventually. 

Thanks for all the help. I really appreciate it. 

Thank you

**Reply by pacmano1** [reply]
Uh, that doesn't make sense at all.   The launcher reaching out should not affect any other processes.

**Reply by JamiB1979** [reply]
Yeah I know....it just creates hundreds of time wait ports and when that happens on a system you can't establish new connections. I see the same traffic that the bug mentions. Do you think I should report it to NextGen?

**Reply by pacmano1** [reply]
What OS?

**Reply by JamiB1979** [reply]
Server 2012 R2..yes I know I've been trying to get these folks to upgrade but small companies don't want to pay for new licenses. At least it's 64 bit! :-)

**Reply by pacmano1** [reply]
I hereby exit the conversation, lol.  I don't run Mirth on windows at all.    If you can, I'd move 'em to Linux.   Even if they have little expertise on Linux, it's mostly a set it and forget it.    But that is another topic of course!

**Reply by JamiB1979** [reply]
I can appreciate that :-) I really am grateful for all your insight.  I'd love to move all our systems to Linux, but the company I work for wouldn't handle that well so alas, I am stuck. 

Thanks again @pacmano1!

**Reply by tonygermano** [reply]
Do you need to run the launcher on the same machine as the server (I'm assuming that's what's happening here?) It's capable of running anywhere as long as the server is reachable. The client OS does not need to match the server OS either.

Updating the local hosts file would prevent all revocation checks for digicert certificates from that machine, not only from the mirth launcher. But hopefully there isn't any casual web browsing being done from the mirth server.

**Reply by JamiB1979** [reply]
Client and server are on the same machine. This is a tiny little hospital that only has one server for our product so it's all done in one. Plus it's also a SQL Server so we try to discourage casual browsing! 

I'm wondering if we are going to hurt remote access with that though. I know what ports RDP uses but they do use a third party service for us to gain access. 

Thanks so much @tonygermano. I'll give that a go and let you know.

**Comment by JamiB1979** [comment, marked as answer]
Thanks for the suggestions folks. What ended up being the best solution for the client is that once the time_wait port time out after 4 mins, we can do what we need, The port exhaustion clears up. Kinda sucks we have to wait 5 mins to stop start a channel, but until NextGen gets it looked at, we'll deal. :-) 

Thanks again for the suggestions.

**Reply by pacmano1** [reply]
I would still review https://docs.microsoft.com/en-us/windows/client-management/troubleshoot-tcpip-port-exhaust and adjust 2012 appropriately.   It describes what you are seeing roughly.
